### DISCUSSION TITLE
Visual Studio Copilot extension - Installation failed

### DISCUSSION THREAD

**Discussion by LuanRykheer** [community/community#11112] [Copilot Conversations] [answered]
### Visual Studio Copilot extension - Installation failed

So I got and email last week saying that I can now test copilot. I think after 4 months of waiting. I followed the README for installing Copilot on VS. In this guide there is a link to to VS extension page for downloading Copilot. 

The first problem I encountred was the installation button, for some reason if I press the installation button nothing happens. I tried refreshing, still did not work. After playing around on the page a bit, I found the version history. This time the link for installing the extension worked. After opening the installer, 

**I get this message:** 
![image](https://user-images.githubusercontent.com/91440879/152714825-d58046fc-8b3d-4e52-9ff1-7ff839186c77.png)

I have **Windows 11** and I have the newest version of VS **(visual studio 2022 professional eddition)**. Other than the version of VS, I can come up with no other reasons for this installation not working.

**Here is the installation log:**
```
  7/02/2022 1:13:55 PM - Microsoft VSIX Installer
  7/02/2022 1:13:55 PM - -------------------------------------------
  7/02/2022 1:13:55 PM - vsixinstaller.exe version:
  7/02/2022 1:13:55 PM - 17.0.5226-preview5
  7/02/2022 1:13:55 PM - -------------------------------------------
  7/02/2022 1:13:55 PM - Command line parameters:
  7/02/2022 1:13:55 PM - C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service\VSIXInstaller.exe,C:\Users\rykhe\Downloads\GitHub.copilot-1.7.4812.vsix
  7/02/2022 1:13:55 PM - -------------------------------------------
  7/02/2022 1:13:55 PM - Microsoft VSIX Installer
  7/02/2022 1:13:55 PM - -------------------------------------------
  7/02/2022 1:13:55 PM - Initializing Install...
  7/02/2022 1:13:55 PM - Extension Details...
  7/02/2022 1:13:55 PM - 	Identifier         : copilot
  7/02/2022 1:13:55 PM - 	Name               : GitHub Copilot
  7/02/2022 1:13:55 PM - 	Author             : GitHub
  7/02/2022 1:13:55 PM - 	Version            : 1.7.4812
  7/02/2022 1:13:55 PM - 	Description        : Your AI pair programmer
  7/02/2022 1:13:55 PM - 	Locale             : en-US
  7/02/2022 1:13:55 PM - 	MoreInfoURL        : 
  7/02/2022 1:13:55 PM - 	InstalledByMSI     : False
  7/02/2022 1:13:55 PM - 	SupportedFrameworkVersionRange : [0.0,2147483647.2147483647]
  7/02/2022 1:13:55 PM - 
  7/02/2022 1:13:55 PM - 	SignatureState     : Unsigned
  7/02/2022 1:13:55 PM - 	Supported Products : 
  7/02/2022 1:13:55 PM - 		Microsoft.VisualStudio.Code
  7/02/2022 1:13:55 PM - 			Version : 
  7/02/2022 1:13:55 PM - 			ProductArchitecture : x86
  7/02/2022 1:13:55 PM - 
  7/02/2022 1:13:55 PM - 	References         : 
  7/02/2022 1:13:55 PM - Signature Details...
  7/02/2022 1:13:55 PM - 	Extension is not signed.
  7/02/2022 1:13:55 PM - 
  7/02/2022 1:13:55 PM - Searching for applicable products...
  7/02/2022 1:13:55 PM - Found installed product - Global Location
  7/02/2022 1:13:55 PM - Found installed product - Visual Studio Professional 2022
  7/02/2022 1:13:55 PM - VSIXInstaller.NoApplicableSKUsException: One or more extensions are for Visual Studio Code. Try installing them in Visual Studio Code.
     at VSIXInstaller.ExtensionService.GetInstallableDataImpl(IInstallableExtension extension, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
     at VSIXInstaller.ExtensionService.GetInstallableData(String vsixPath, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
     at VSIXInstaller.ExtensionPackService.IsExtensionPack(IStateData stateData, Boolean isRepairSupported)
     at VSIXInstaller.ExtensionPackService.ExpandExtensionPackToInstall(IStateData stateData, Boolean isRepairSupported)
     at VSIXInstaller.App.Initialize(Boolean isRepairSupported)
     at VSIXInstaller.App.Initialize()
     at System.Threading.Tasks.Task`1.InnerInvoke()
     at System.Threading.Tasks.Task.Execute()
  --- End of stack trace from previous location where exception was thrown ---
     at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```
**PS:** After looking through the installation log, if I understand correctly it says I should try installing from within VS. I have tried this. I cannot find Copilot on the extions page in VS.  Also this error is the same across all avalible version of Copilot.

Please help.

**Comment by Kek5chen** [comment, marked as answer]
You are trying to install Copilot to Visual Studio. As of right now it is only available for Visual Studio **Code**. It also says that in the VSIX installer.

**Reply by LuanRykheer** [reply]
Thanks, 
I didn't even know Visual Studio Code was different from Visual Studio. The installation worked.

**Comment by musaffarpatel** [comment]
The following articles suggests is supported for Visual Studio:

https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-visual-studio

However, I am having the same problem as LuanRykheer.  Installing on Visual Studio 17.4 on Windows 11

**Reply by Kek5chen** [reply]
GitHub copilot is only supported on VS 2022

**Reply by davidmitten** [reply]
Description        : GitHub Copilot Completions is an AI pair programmer that helps you write code faster and with less work. For Visual Studio **17.10** or higher

**Reply by davidmitten** [reply]
It doesn't seem to work for Visual Studio Community Edition though?

**Reply by Kek5chen** [reply]
@davidmitten you do know that this question is 2+ years old, right?
